invenio_oaiserver/tasks.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Task for OAI."""
+
+from celery import shared_task
+from flask_celeryext import RequestContextTask
+from invenio_db import db
+from invenio_records.api import Record
+
+from .query import get_affected_records
+
+
+@shared_task(base=RequestContextTask)
+def update_records_after_change_oaiset(spec=None, search_pattern=None):
+    """Update records after a oaiset is updated, inserted or removed."""
+    for record_id in get_affected_records(spec=spec,
+                                          search_pattern=search_pattern):
+        record = Record.get_record(id=record_id)
+        record.commit()
+        db.session.commit()
```

Comment:
IMHO we should have a transaction around and commit after all records are updated.